addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/ParamSinksUtils.java

```diff
@@ -109,19 +91,19 @@ private static void setSinkForSource(UserDataSource source, HttpMessage sinkMsg)
                             + sinkMsg.getRequestHeader().getURI());
         }
         HashSet<Integer> sinks = sourceToSinks.get(source.toString());
+        HashSet<Integer> sinksHashCodes = sourceToSinksHashCodes.get(source.toString());
         if (sinks == null) {
             sinks = new HashSet<>();
+            sinksHashCodes = new HashSet<>();
         }
-        try {
-            HistoryReference hRef =
-                    new HistoryReference(
-                            Model.getSingleton().getSession(),
-                            HistoryReference.TYPE_SCANNER_TEMPORARY,
-                            sinkMsg);
-            sinks.add(Integer.valueOf(hRef.getHistoryId()));
+
+        int msgHashCode = sinkMsg.hashCode();
+        if (!sinksHashCodes.contains(msgHashCode)) {
```

Comment:
a better way to achieve the same is using `sinksHashCodes.add()` returns boolean so depend on that instead of manually doing the same.

Reply:
I prefer as it is because it expresses better what I want to do. If you want I can change It.

Reply:
ok i am ok with both ways,  lets wait for @kingthorin @thc202 to comment.

Reply:
I'm good either way unless there's some concern about performance.